python/ray/serve/tests/test_proxy_actor_wrapper.py

```diff
@@ -0,0 +1,242 @@
+import asyncio
+import concurrent.futures
+from unittest.mock import Mock
+
+import pytest
+
+from ray.exceptions import RayTaskError
+from ray.serve._private.proxy_state import ActorProxyWrapper, wrap_as_future
+from ray.serve.schema import LoggingConfig
+
+
+def _create_object_ref_mock():
+    fut = concurrent.futures.Future()
+
+    ray_object_ref_mock = Mock(future=Mock(return_value=fut))
+
+    return ray_object_ref_mock, fut
+
+
+def _create_mocked_actor_proxy_wrapper(actor_handle_mock):
+    return ActorProxyWrapper(
+        logging_config=LoggingConfig(),
+        actor_handle=actor_handle_mock,
+        node_id="some_node_id",
+    )
+
+
+@pytest.mark.asyncio
+async def test_wrap_as_future_timeout():
+    object_ref_mock, fut = _create_object_ref_mock()
+
+    aio_fut = wrap_as_future(ref=object_ref_mock, timeout_s=0)
+
+    assert not aio_fut.done()
+    # Yield the event-loop
+    await asyncio.sleep(0.001)
+
+    assert aio_fut.done()
+    with pytest.raises(TimeoutError) as exc_info:
+        aio_fut.result()
+
+    assert "Future cancelled after timeout 0s" in str(exc_info.value)
+
+
+@pytest.mark.asyncio
+async def test_wrap_as_future_success():
+    # Test #1: Validate wrapped asyncio future completes, upon completion of the
+    #          ObjectRef's one
+
+    object_ref_mock, fut = _create_object_ref_mock()
+
+    aio_fut = wrap_as_future(ref=object_ref_mock, timeout_s=3600)
+
+    assert not aio_fut.done()
+    # Complete source future (ObjectRef one)
+    fut.set_result("test")
+    # Yield the event-loop
+    await asyncio.sleep(0.001)
+
+    assert aio_fut.done()
+    assert aio_fut.result() == "test"
```

Comment:
You can use `ray._private.test_utils.async_wait_for_condition` as a common pattern for adding these assertions that require the loop to be yielded